### ISSUE TITLE
[Issue] GH CLI login loop in DevContainer

### ISSUE THREAD

**Issue by jongio** [Azure/azure-dev#76]
- [ ] Make sure you've installed the latest version using [instructions in the wiki](../wiki/install)

**Output from `azd version`**
Run `azd version` and copy and paste the output here:

**Output from `az version`** 
Run `az version` and copy and paste the output here (minimum required version is 2.38.0): 

**Describe the bug**
Description of issue you're seeing...

stuck in gh login loop after calling azd pipeline config

```
? This command requires you to be logged into GitHub. Log in using the GitHub CLI? Yes
? What is your preferred protocol for Git operations? HTTPS
? How would you like to authenticate GitHub CLI? Login with a web browser

! First copy your one-time code: 
Press Enter to open github.com in your browser... 
failed to authenticate via web browser: non-200 OK status code: 401 Unauthorized body: "{\"message\":\"Bad credentials\",\"documentation_url\":\"https://docs.github.com/graphql\"}"
There was an issue logging into GitHub.
? This command requires you to be logged into GitHub. Log in using the GitHub CLI? Yes
? What is your preferred protocol for Git operations? HTTPS
? How would you like to authenticate GitHub CLI? Login with a web browser

! First copy your one-time code: 
Press Enter to open github.com in your browser... 
failed to authenticate via web browser: non-200 OK status code: 401 Unauthorized body: "{\"message\":\"Bad credentials\",\"documentation_url\":\"https://docs.github.com/graphql\"}"
There was an issue logging into GitHub.
? This command requires you to be logged into GitHub. Log in using the GitHub CLI? No
Error: failed to ensure login to GitHub: interactive GitHub login declined; use `gh login` to log into GitHub
```

**To Reproduce**
Steps to reproduce the behavior...

open project in dev container
run azd pipeline config
you'll get to the gh login prompt and see this 

**Expected behavior**

gh login

**Environment**
Information on your environment:
    * Language name and version
    * IDE and version : [e.g. Visual Studio 16.3]

devcontainer

**Additional context**

I'm also seeing the 401 when I call `gh auth login` directly.

**Comment by jongio** [comment]
Workaround is to use PAT when logging in instead of browser.

**Comment by jongio** [comment]
Known GH CLI issue: https://github.com/cli/cli/issues/5927 PR is in progress. https://github.com/cli/cli/pull/5928

**Comment by jongio** [comment]
gh CLI PR has been merged. I'm testing out devcontainer now

**Comment by jongio** [comment]
Confirming this was fixed in gh@2.14.1

If you see this issue, then Rebuild the devcontainer without cache to get the latest gh cli
